src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -937,6 +937,11 @@ void NetworkOPsImp::processTransaction (std::shared_ptr<Transaction>& transactio
         doTransactionSync (transaction, bUnlimited, failType);
     else
         doTransactionAsync (transaction, bUnlimited, failType);
+
+    auto const lpClosed = m_ledgerMaster.getValidatedLedger ();
+
+    if (lpClosed)
+        transaction->setValidatedLedgerIndex(lpClosed->info().seq);
```

Comment:
Alternate suggestion:
```
if (auto const valLedger {m_ledgerMaster.getValidatedLedger()}; valLedger)
    transaction->setValidatedLedgerIndex(valLedger->info().seq);```

Reply:
Or even just
```
    if (auto const l = m_ledgerMaster.getValidatedLedger())
        transaction->setValidatedLedgerIndex(l->info().seq);
```
I.e. there's no need for the `; valLedger)` part.

Reply:
Can we set this in `apply` instead? That way all the auxiliary ledger information is set in the same place.

Reply:
Done, simplified, moved to apply.